### PR TITLE
fix(manager/conan): allow ref with user only

### PR DESCRIPTION
**Pull request by dazze1812**
## Changes

For Conan package references "user" can be used without specifying a "channel".

Now `package/1.2.3@user` is also possible.

## Context

see https://github.com/renovatebot/renovate/discussions/26040

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
Please test on a real repo to ensure the `conan` datasource can also handle this.

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/conan/extract.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import type { PackageDependency, PackageFileContent } from '../types';
 import { isComment } from './common';
 
 const regex = regEx(
-  `(?<name>[-_a-z0-9]+)/(?<version>[^@\n{*"']+)(?<userChannel>@[-_a-zA-Z0-9]+/[^#\n.{*"' ]+)?#?(?<revision>[-_a-f0-9]+[^\n{*"'])?`,
+  `(?<name>[-_a-z0-9]+)/(?<version>[^@\n{*"']+)(?<userChannel>@[-_a-zA-Z0-9]+(/[^#\n.{*"' ]+|))?#?(?<revision>[-_a-f0-9]+[^\n{*"'])?`,
```
Comment (viceice) [inline lib/modules/manager/conan/extract.ts:7]:
```suggestion
  `(?<name>[-_a-z0-9]+)/(?<version>[^@\n{*"']+)(?<userChannel>@[-_a-zA-Z0-9]+(?:/[^#\n.{*"' ]+|))?#?(?<revision>[-_a-f0-9]+[^\n{*"'])?`,
```

**Comment by viceice** [comment]
> Please test on a real repo to ensure the `conan` datasource can also handle this.

☝️

**Comment by dazze1812** [comment]
> > Please test on a real repo to ensure the `conan` datasource can also handle this.
> 
> ☝️

Sorry, but I still have no test environment. To test this for our use case I need a test on our local Conan and GitLab infrastructure. Is there a guide line how to setup a local test? I only found the guide line for testing on a public GitHub repo.

**Comment by rarkins** [comment]
Locally you can run with `pnpm start ....` instead of `renovate ....`. All other config (e.g. platform, endpoint, token, etc) remain the same

File: lib/modules/versioning/conan/common.ts [hunk 2]
```diff
@@ -30,7 +30,12 @@ export function makeVersion(
 export function cleanVersion(version: string): string {
   if (version) {
     return version
-      .replace(regEx(/,|\[|\]|"|include_prerelease=|loose=|True|False/g), '')
+      .replace(
```
Comment (viceice) [inline lib/modules/versioning/conan/common.ts:33]: please do these changes to versioning in a separate PR

Comment (dazze1812) [reply lib/modules/versioning/conan/common.ts:33]: Done -> https://github.com/renovatebot/renovate/pull/26588

**Comment by viceice** [comment]
> Please test on a real repo to ensure the `conan` datasource can also handle this.

where did you test this?

**Comment by dazze1812** [comment]
> > Please test on a real repo to ensure the `conan` datasource can also handle this.
> 
> where did you test this?

We have an internal GitLab and Artifactory installation. The public Conan center packages do not have the user attribute so I needed to test it internally.

**Review by viceice — APPROVED** [review]
